skylog/storage/cloudspanner/schema.sdl

```diff
@@ -0,0 +1,9 @@
+-- TreeNodes stores Merkle tree nodes. The nodes are sharded according to
+-- ShardId, which is computed based on the tree shape and write pattern, in
+-- order to maximize writing parallelism.
+CREATE TABLE TreeNodes (
+  TreeId       INT64 NOT NULL,
+  ShardId      INT32 NOT NULL,
```

Comment:
Looks like there could be a key hotspot here if I read the docs attached to the code correctly. Not sure about the plans for splits though.

Reply:
OK but be aware that the order fields appear in the primary key defines the physical data layout and IIRC you can only split on a key prefix. When there are multiple trees how do we want the split to work? TreeId, ShardId seems plausible but if we wanted to split by ShardId then they'd probably need to be swapped round.

Just for my interest for reads do you know roughly how many shards a proof path is expected to cross?

Reply:
I'm fairly sure that splits can be done by (TreeID, ShardID) pairs, but I will double check once I get to test it.

Reply:
A proof path will probably cross about 2 shards - one for levels below `ShardLevels`, and one above.

Reply:
Yep I think if you use this layout you can split by TreeID, ShardID but not by ShardID on it's own.

Overall, I think it's OK.